### PR TITLE
FIX: Redirect old Loading data in PyTorch to newer Datasets & DataLoaders

### LINKED ISSUE #2905: Test the Loading data in PyTorch tutorial
Test the following tutorial:
 https://pytorch.org/tutorials/recipes/recipes/loading_data_recipe.html
 **Follow these steps:**
1. Fork the pytorch/tutorials repo.
2. Switch to a new branch.
3. Run the tutorial as a python script:
```
python3 <path-to-tutorial>
```
4. Save the output into a .txt file. Did you see any errors? Add them to the output.
5. Run the tutorial in the Google Colab and save the outputs into a .txt file. Did you see any issues? Take screenshots of the issues.
6. Did the tutorial make sense? Can you suggest any improvements? Past all the outputs, screenshots, and suggestions for improvements in the issue and ping one of the reviewers. If your suggestion is approved, please submit a PR with the fix.

cc @svekars @kit1980

### PR DESCRIPTION
**Pull request by loganthomas**
Fixes #2905 

Followed the above guidance to deprecate old and point to new content

## Checklist

- [X] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [X] Only one issue is addressed in this pull request
- [X] Labels from the issue that this PR is fixing are added to this pull request
- [X] No unnecessary issues are included into this pull request.

cc @sekyondaMeta @svekars @kit1980 @brycebortree

### CODE REVIEW

File: .jenkins/validate_tutorials_built.py [hunk 1]
```diff
@@ -38,7 +38,7 @@
     "prototype_source/nestedtensor",
     "recipes_source/recipes/saving_and_loading_models_for_inference",
     "recipes_source/recipes/saving_multiple_models_in_one_file",
-    "recipes_source/recipes/loading_data_recipe",
+    # "recipes_source/recipes/loading_data_recipe",  # deprecated and replaced with beginner_source/basics/data_tutorial.py
```
Comment (svekars) [inline .jenkins/validate_tutorials_built.py:41]: just delete line 41.

File: recipes_source/recipes_index.rst [hunk 2]
```diff
@@ -407,7 +407,7 @@ Recipes are bite-sized, actionable examples of how to use specific PyTorch featu
 .. toctree::
    :hidden:
 
-   /recipes/recipes/loading_data_recipe
+   /recipes/loading_data_recipe
```
Comment (svekars) [inline recipes_source/recipes_index.rst:410]: since we are deleting it, just remove line 410

File: recipes_source/recipes_index.rst [hunk 3]
```diff
@@ -34,7 +34,7 @@ Recipes are bite-sized, actionable examples of how to use specific PyTorch featu
    :header: Loading data in PyTorch
    :card_description: Learn how to use PyTorch packages to prepare and load common datasets for your model.
    :image: ../_static/img/thumbnails/cropped/loading-data.PNG
-   :link: ../recipes/recipes/loading_data_recipe.html
+   :link: ../beginner/basics/data_tutorial.html
```
Comment (svekars) [inline recipes_source/recipes_index.rst:37]: since we are removing it, just delete line 37.

File: recipes_source/recipes/README.txt [hunk 4]
```diff
@@ -2,7 +2,7 @@ PyTorch Recipes
 ---------------------------------------------
 1. loading_data_recipe.py
 	 Loading Data in PyTorch
-         https://pytorch.org/tutorials/recipes/recipes/loading_data_recipe.html
+         https://pytorch.org/tutorials/recipes/loading_data_recipe.html
```
Comment (svekars) [inline recipes_source/recipes/README.txt:5]: just delete this entry.

File: recipes_source/recipes_index.rst (hunk 2 above)
Comment (loganthomas) [reply recipes_source/recipes_index.rst:410]: I thought the ask was to deprecating and **creating a redirect from this tutorial to the data_tutorial.html**. Wouldn't this remove the tutorial entirely without a redirect?

File: recipes_source/recipes_index.rst (hunk 3 above)
Comment (loganthomas) [reply recipes_source/recipes_index.rst:37]: I thought the ask was to deprecating and **creating a redirect from this tutorial to the data_tutorial.html**. Wouldn't this remove the tutorial entirely without a redirect?

File: recipes_source/recipes/README.txt (hunk 4 above)
Comment (loganthomas) [reply recipes_source/recipes/README.txt:5]: Shouldn't this point to the Datasets & DataLoaders tutorial instead of being removed completely?

File: recipes_source/recipes_index.rst (hunk 3 above)
Comment (svekars) [reply recipes_source/recipes_index.rst:37]: No, it will still convert that .rst to html but, you can add [:orphan:](https://www.sphinx-doc.org/en/master/usage/restructuredtext/field-lists.html#metadata) at the top of the .rst page so Sphinx doesn't drop a warning in the log. This whole card needs to go

File: recipes_source/recipes_index.rst (hunk 2 above)
Comment (svekars) [reply recipes_source/recipes_index.rst:410]: same as above.

File: recipes_source/recipes/README.txt (hunk 4 above)
Comment (svekars) [reply recipes_source/recipes/README.txt:5]: No, this file needed for sphinx gallery. You can delete that entry.

**Comment by loganthomas** [comment]
thank you @svekars 🎉
